subsys/bluetooth/host/l2cap.c

```diff
@@ -589,24 +591,47 @@ static struct bt_l2cap_server *l2cap_server_lookup_psm(u16_t psm)
 
 int bt_l2cap_server_register(struct bt_l2cap_server *server)
 {
-	if (server->psm < L2CAP_LE_PSM_START ||
-	    server->psm > L2CAP_LE_PSM_END || !server->accept) {
+	if (!server->accept) {
 		return -EINVAL;
 	}
 
+	if (server->psm) {
+		if (server->psm < L2CAP_LE_PSM_FIXED_START ||
```

Comment:
We do use bt_l2cap_server_register for registering fixed psm like in case of IPSP,  I guess we can remove this check an let the application register fixed PSMs.